### ISSUE TITLE
latexmlpost dies with "bad ns attribute!"

### ISSUE THREAD

**Issue by epifanovsky** [brucemiller/LaTeXML#1286]
I am running latexml as supplied with Ubuntu 20.04. Specifically:
```
$ perl --version

This is perl 5, version 26, subversion 1 (v5.26.1) built for x86_64-linux-gnu-thread-multi

$ latexml  --VERSION
latexml (LaTeXML version 0.8.2)
```

It appears that the error occurs when the latex file contains bibliography and math mode in section titles. Here is one example:
```
\documentclass{article}
\begin{document}
\tableofcontents
\section{Hello $x=2$}
\bibliographystyle{unsrt}
\begin{thebibliography}{9}
\end{thebibliography}
\end{document}
```

Steps to reproduce:
```
$ latexml -o bug.xml bug.tex
$ latexmlpost bug.xml
```

And here is the observed output:
```
$ latexml -o bug.xml bug.tex
latexml (LaTeXML version 0.8.2)
processing started Fri Jun 12 18:41:38 2020

(Digesting TeX bug...
(Loading /usr/share/perl5/LaTeXML/Package/TeX.pool.ltxml...
(Loading /usr/share/perl5/LaTeXML/Package/eTeX.pool.ltxml... 0.00 sec)
(Loading /usr/share/perl5/LaTeXML/Package/pdfTeX.pool.ltxml... 0.00 sec) 0.07 sec)
(Processing content /tmp/latexml/bug.tex...
(Loading /usr/share/perl5/LaTeXML/Package/LaTeX.pool.ltxml... 0.09 sec)
(Loading /usr/share/perl5/LaTeXML/Package/article.cls.ltxml... 0.01 sec) 0.12 sec) 0.20 sec)
(Building...
(Loading compiled schema /usr/share/perl5/LaTeXML/resources/RelaxNG/LaTeXML.model... 0.01 sec). 0.03 sec)
(Rewriting... 0.00 sec)
(Math Parsing...1 formulae ...[1]
Math parsing succeeded:
   ltx:XMath: 1/1
Symbols assumed as simple identifiers (with # of occurences):
   'x{italic}' (1)
Set MATHPARSER_SPECULATE to speculate on possible notations.
 0.01 sec)
(Finalizing... 0.00 sec)
Conversion complete: No obvious problems.
processing finished Fri Jun 12 18:41:38 2020

$ latexmlpost bug.xml
latexmlpost (LaTeXML version 0.8.2)
processing started Fri Jun 12 18:42:08 2020
(post-processing...
(Scan  processing... 0.00 sec)
(MakeBibliography  processing... [using bibliographies ] [0 bibentries, 0 cited] 0.01 sec)
(CrossRef  processing...
Fatal:perl:die Perl died
        bad ns attribute! at /usr/lib/x86_64-linux-gnu/perl5/5.26/XML/LibXML.pm line 1690.
        In Post::Document[] ->addNodes
         <= Post::CrossRef[@0x55aac44b6e00]
1 fatal error

Postprocessing failed: 1 fatal error
processing abandoned Fri Jun 12 18:42:08 2020
```
```

**Comment by dginev** [comment]
Hi @epifanovsky , one of the curious pieces of your report is that you are running a substantially older version of latexml (0.8.2). We are currently at 0.8.4 and are soon to release 0.8.5, so this may make it harder to resolve your issue for your specific installation.

I tried your example with the latest 0.8.4 master of latexml, and all seemed to be in order. Could you try updating your latexml and double-checking if that works better?

**Comment by epifanovsky** [comment]
Thanks for the quick response @dginev. Sorry, I indicated that I was running Ubuntu 20.04 but collected the outputs on an older version. The issue is still there with latexml 0.8.4 as packaged with Ubuntu 20.04:
```
$ latexml -o bug.xml bug.tex
latexml (LaTeXML version 0.8.4)
processing started Fri Jun 12 18:58:27 2020

(Digesting TeX bug...
(Loading /usr/share/perl5/LaTeXML/Package/TeX.pool.ltxml...
(Loading /usr/share/perl5/LaTeXML/Package/eTeX.pool.ltxml... 0.00 sec)
(Loading /usr/share/perl5/LaTeXML/Package/pdfTeX.pool.ltxml... 0.00 sec) 0.08 sec)
(Processing content /tmp/latexml/bug.tex...
(Loading /usr/share/perl5/LaTeXML/Package/LaTeX.pool.ltxml... 0.10 sec)
(Loading /usr/share/perl5/LaTeXML/Package/article.cls.ltxml... 0.01 sec) 0.14 sec) 0.22 sec)
(Building...
(Loading compiled schema /usr/share/perl5/LaTeXML/resources/RelaxNG/LaTeXML.model... 0.01 sec). 0.04 sec)
(Rewriting... 0.00 sec)
(Math Parsing...1 formulae ...[1]
Math parsing succeeded:
   ltx:XMath: 1/1
Symbols assumed as simple identifiers (with # of occurences):
   'x{italic}' (1)
Set MATHPARSER_SPECULATE to speculate on possible notations.
 0.01 sec)
(Finalizing... 0.00 sec)
Conversion complete: No obvious problems.
processing finished Fri Jun 12 18:58:28 2020

$ latexmlpost bug.xml
latexmlpost (LaTeXML version 0.8.4)
processing started Fri Jun 12 18:58:36 2020
(post-processing...
(Scan  processing... 0.00 sec)
(MakeBibliography  processing... [using bibliographies ] [0 bibentries, 0 cited] 0.01 sec)
(CrossRef  processing...
Fatal:perl:die Perl died
        bad ns attribute! at /usr/lib/x86_64-linux-gnu/perl5/5.30/XML/LibXML.pm line 1690.
        In Post::Document[] ->addNodes
         <= Post::CrossRef[@0x564ac46cd450]
1 fatal error

Postprocessing failed: 1 fatal error
processing abandoned Fri Jun 12 18:58:36 2020
```

Let me try to pull the latest version from this repository and try again.

**Comment by dginev** [comment]
Hope it works with the latest github master (it does for me). We added a lot of robustness work this spring in stress-testing with the latest arXiv articles, so I am hoping this error was resolved along the way as well. We definitely had some namespace-related troubles, though my memory is a bit vague.

If indeed the packaged 0.8.4 contains this fatal, we should use this as extra incentive to start working on 0.8.5.

**Comment by epifanovsky** [comment]
Okay, I can confirm now that it works as expected with current master. Thanks again for the quick responses!

**Comment by dginev** [comment]
Sure, thanks for letting us know of the issue!

**Comment by brucemiller** [comment]
I think this was fixed in #1247, back in March.
But yeah, motivation for a release :>
